package.json

```diff
@@ -28,11 +28,12 @@
   "scripts": {
     "predev": "npm install && npx update-browserslist-db@latest",
     "dev": "npx vuepress dev",
-    "prepare-docs": "ruby prepare-docs.rb",
+    "prepare-docs": "ruby prepare-docs.rb $ARGUMENTS",
     "add-blog-meta": "ruby add-blog-meta.rb",
     "build": "ruby prepare-docs.rb && ruby add-blog-meta.rb && vuepress build .",
     "build-only": "vuepress build .",
-    "build-local": "rm -f javadoc-*.tgz* && npm run prepare-docs && npm run dev",
+    "prebuild-local": "npm run prepare-docs",
```

Comment:
Actually the `pre` prefix is a convention of `npm` for a script that is executed automatically as a pre requisit of another step (in that case `build-local`). Therefore it is an internal script that normally is not called by the user.

the `json` specification does not allow comments by default - but most editors can be configured to accept them (as there are many opinionated discussions about this topic). Unfortunately many tools show those scripts as all the others. I can document this - but this is a common mechanism - so I tend to document only the scripts that should be used by the users. I see several options:

- keep as is
- document the magic (that's my least prefered one)
- use `&&` to combine this step with the `build-local` (as it is done in the `build` script) - but if I remember correctly, with this I had problems to pass arguments properly from the command line to the local build setp (therefore also not recommended)
- create a helper script that is called, that will do the necessary steps (but this might be problematic on different platforms - but git bash on windows should be able to handle this) 

Let me know what you think.

Reply:
Hmm seems the last time I worked with npm scripts was too much time ago, now that you say it, I do remember the pre and post scripts.

I would vote for keep as is 👍